File: v1/src/main/java/com/api/v1/domain/Rent.java
package com.api.v1.domain;

import lombok.Getter;
import org.springframework.data.annotation.Id;
import org.springframework.data.mongodb.core.mapping.Document;
import org.springframework.data.mongodb.core.mapping.Field;

import java.math.BigDecimal;
import java.math.BigInteger;
import java.time.ZonedDateTime;
import java.util.UUID;

@Getter
@Document(collection = "v1_rents")
public class Rent {

    @Id
    private UUID id;

    @Field
    private BigInteger orderNumber;

    @Field
    private Customer customer;

    @Field
    private Car car;

    @Field
    private Payment payment;

    @Field
    private int days;

    @Field
    private BigDecimal price;

    @Field
    private String returningDate;

    @Field
    private String returnedAt;

    @Field
    private String rentedAt;

    public void terminate() {
        this.rentedAt = ZonedDateTime.now().toString();
    }

}

File: v2/src/main/kotlin/com/api/v2/rent/domain/Rent.kt
package com.api.v2.rent.domain

import com.api.v2.car.domain.Car
import com.api.v2.customer.domain.Customer
import com.api.v2.payment.domain.Payment
import com.api.v2.payment.utils.PaymentOrderGeneratorUtil
import org.springframework.data.annotation.Id
import org.springframework.data.mongodb.core.mapping.Document
import java.math.BigInteger
import java.time.ZonedDateTime
import java.util.*

@Document(collection = "v2_rents")
data class Rent(
    @Id val id: UUID,
    val orderNumber: BigInteger,
    val customer: Customer,
    val car: Car,
    val payment: Payment,
    val rentedAt: String
) {

    constructor(
        customer: Customer,
        car: Car,
        payment: Payment
    ) : this(
        UUID.randomUUID(),
        PaymentOrderGeneratorUtil.generate(),
        customer,
        car,
        payment,
        ZonedDateTime.now().toString()
    )

}